splash/har/qt.py

```diff
@@ -129,16 +133,17 @@ def reply2har(reply, include_content=False, binary_content=False):
         res["redirectURL"] = ""
 
     if include_content:
-        data = bytes(reply.readAll())
-        if binary_content:
-            res["content"]["encoding"] = "binary"
-            res["content"]["text"] = data
-            res["content"]["size"] = len(data)
-        else:
-            res["content"]["encoding"] = "base64"
-            res["content"]["text"] = base64.b64encode(data)
-            res["content"]["size"] = len(data)
-
+        content = getattr(reply, 'content', None)
+        if content is not None:
+            res["content"]["size"] = content.size()
+            bytes_array = bytes(content)
+
+            if not is_binary_string(bytes_array):
+                res["content"]["text"] = bytes_array.decode('utf8', 'replace')
```

Comment:
I think we shouldn't try to detect if data is binary ourselves. `.decode('utf8', 'replace')` can lead to data loss if information doesn't look like binary, but can't be decoded using utf8. It also means that client may occasionally get a result which is not base64-encoded (if value which is usually binary only contain ascii letters) - it means client will have to always handle both base64 and raw cases. Currently it is enough to handle base64 case.